vic/drivers/cesm/cpl_mct/lnd_comp_mct.F90

```diff
@@ -0,0 +1,738 @@
+    LOGICAL                                     :: exists  ! true if file exists
+    CHARACTER(len=*), PARAMETER                 :: subname = '(lnd_init_mct)'
+
+    !--- Local Variables for C/Fortran Interface
+    INTEGER(C_INT)                              :: errno
+    CHARACTER(len=VICMAXSTRING, kind=C_CHAR)    :: vic_global_param_file
+
+    !--- initialize the VIC logging to stderr (for now)
+    CALL initialize_log()
+
+    !--- initialize the field index values, subroutine is below in this file
+    CALL cpl_indices_set()
+
+    !--- point/get data from cdata datatype
+    CALL seq_cdata_setptrs(cdata, ID=LNDID, mpicom=mpicom_lnd, &
+         gsMap=GSMap_lnd, dom=dom_lnd, infodata=infodata)
+    CALL MPI_COMM_RANK(mpicom_lnd, mytask, ierr)
+
+    !--- copy/hand the mpicom from the driver to the vic model
+    CALL initialize_vic_cesm_mpi(mpicom_lnd)
+
+    !--- get unit number for vic log file
+    !--- setup vic log file
+    !--- set shr unit number to vic log file
+    CALL shr_file_getLogUnit(shrlogunit)
+    IF (mytask == 0) THEN
+       iulog = shr_file_getUnit()
+       CALL shr_file_setIO('lnd_modelio.nml', iulog)
+       WRITE(iulog, *) subname,': VIC land model initialization'
+    ELSE
+       iulog = shrlogunit
+    ENDIF
+    CALL set_print_library_iulog(iulog)
+    CALL shr_file_setLogUnit(iulog)
+
+    !--- get the casename, needed for output file names
+    CALL seq_infodata_GetData(infodata, case_name=caseid)
+
+    !--- get the starttype, seq_infodata_start_type_[start,cont,brnch] = clean start, restart, branch
+    CALL seq_infodata_GetData(infodata, start_type=starttype)
+
+    !--- get other metadata
+    CALL seq_infodata_GetData(infodata, case_desc=casedesc, model_version=version, &
+                              hostname=hostname, username=username)
+
+    cmeta%caseid = Copy_s2a(caseid)
+    cmeta%casedesc = Copy_s2a(casedesc)
+    cmeta%starttype = Copy_s2a(starttype)
+    cmeta%model_version = Copy_s2a(version)
+    cmeta%hostname = Copy_s2a(hostname)
+    cmeta%username = Copy_s2a(username)
+
+    !--- get some clock info
+    CALL seq_timemgr_EClockGetData(EClock, &
+         dtime=dtime, &
+         curr_yr=year, &
+         curr_mon=month, &
+         curr_day=day, &
+         curr_tod=seconds, &
+         calendar=calendar)
+
+    !--- initialize clock and add starting date / time info
+    vclock%timestep = dtime
+    vclock%current_year = year
+    vclock%current_month = month
+    vclock%current_day = day
+    vclock%current_dayseconds = seconds
+    vclock%state_flag = .FALSE.
+    vclock%stop_flag = .FALSE.
+    vclock%calendar = Copy_s2a(calendar)
+
+    !--- VIC global parameter file (namelist)
+    ! commenting out because it was yielding a drv_in <-- need to rethink this
+    !IF(PRESENT(NLFilename)) THEN
+    !   vic_global_param_file = NLFilename
+    !ELSE
+    !   vic_global_param_file = 'vic.globalconfig.txt'
+    !ENDIF
+    vic_global_param_file = 'vic.globalconfig.txt'
```

Comment:
I think we do, at least for now. This filename matches that from `build_vic_namelist`.  I don't think CESM has any utility for passing our driver a namelist filename. Basically, we use `build_vic_namelist` to copy `vic.globalconfig.txt` to the run directory and we use it directly from there. This is how the other components do it, although they have their own names for their namelists.